service/src/main/resources/db/migration/V438__assistance_need_decision_no_end_date.sql

```diff
@@ -0,0 +1,7 @@
+ALTER TABLE assistance_need_decision
+    ADD COLUMN end_date_not_known BOOLEAN DEFAULT FALSE NOT NULL,
+    ADD CONSTRAINT check$assistance_need_decision_validity_period_end_date
+        CHECK (
+            'ASSISTANCE_SERVICES_FOR_TIME' <> ANY(assistance_levels) OR
+            (end_date_not_known = FALSE OR upper(validity_period) IS NULL)
+            );
```

Comment:
I think there's some error in the logic here. For example it always allows end date to be missing irrespective of the other conditions. All these three conditions are combined with OR so currently it's enough that any one of them is TRUE.